pkg/prometheus/statefulset.go

```diff
@@ -325,222 +351,30 @@ func makeStatefulSetSpec(
-		}
-
-		if p.Spec.Query.MaxSamples != nil && *p.Spec.Query.MaxSamples > 0 {
-			promArgs = cg.WithMinimumVersion("2.5.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "query.max-samples", Value: fmt.Sprintf("%d", *p.Spec.Query.MaxSamples)})
-		}
-
-		if p.Spec.Query.MaxConcurrency != nil && *p.Spec.Query.MaxConcurrency > 1 {
-			promArgs = append(promArgs, monitoringv1.Argument{Name: "query.max-concurrency", Value: fmt.Sprintf("%d", *p.Spec.Query.MaxConcurrency)})
-		}
-
-		if p.Spec.Query.Timeout != nil {
-			promArgs = append(promArgs, monitoringv1.Argument{Name: "query.timeout", Value: string(*p.Spec.Query.Timeout)})
-		}
-	}
-
-	if p.Spec.Web != nil {
-		if p.Spec.Web.PageTitle != nil {
-			promArgs = cg.WithMinimumVersion("2.6.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "web.page-title", Value: *p.Spec.Web.PageTitle})
-		}
-
-		if p.Spec.Web.MaxConnections != nil {
-			promArgs = append(promArgs, monitoringv1.Argument{Name: "web.max-connections", Value: fmt.Sprintf("%d", *p.Spec.Web.MaxConnections)})
-		}
-	}
-
-	if p.Spec.EnableAdminAPI {
-		promArgs = append(promArgs, monitoringv1.Argument{Name: "web.enable-admin-api"})
-	}
-
-	if p.Spec.EnableRemoteWriteReceiver {
-		promArgs = cg.WithMinimumVersion("2.33.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "web.enable-remote-write-receiver"})
-	}
-
-	if len(p.Spec.EnableFeatures) > 0 {
-		promArgs = cg.WithMinimumVersion("2.25.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "enable-feature", Value: strings.Join(p.Spec.EnableFeatures[:], ",")})
-	}
-
-	if p.Spec.ExternalURL != "" {
-		promArgs = append(promArgs, monitoringv1.Argument{Name: "web.external-url", Value: p.Spec.ExternalURL})
-	}
-
 	webRoutePrefix := "/"
-	if p.Spec.RoutePrefix != "" {
-		webRoutePrefix = p.Spec.RoutePrefix
-	}
-	promArgs = append(promArgs, monitoringv1.Argument{Name: "web.route-prefix", Value: webRoutePrefix})
-
-	if p.Spec.LogLevel != "" && p.Spec.LogLevel != "info" {
-		promArgs = append(promArgs, monitoringv1.Argument{Name: "log.level", Value: p.Spec.LogLevel})
-	}
-
-	if p.Spec.LogFormat != "" && p.Spec.LogFormat != "logfmt" {
-		promArgs = cg.WithMinimumVersion("2.6.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "log.format", Value: p.Spec.LogFormat})
-	}
-
-	if p.Spec.WALCompression != nil {
-		arg := monitoringv1.Argument{Name: "no-storage.tsdb.wal-compression"}
-		if *p.Spec.WALCompression {
-			arg.Name = "storage.tsdb.wal-compression"
-		}
-		promArgs = cg.WithMinimumVersion("2.11.0").AppendCommandlineArgument(promArgs, arg)
-	}
-
-	if p.Spec.AllowOverlappingBlocks {
-		promArgs = cg.WithMinimumVersion("2.11.0").WithMaximumVersion("2.39.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "storage.tsdb.allow-overlapping-blocks"})
+	if cpf.RoutePrefix != "" {
+		webRoutePrefix = cpf.RoutePrefix
```

Comment:
What if we flip the condition and then we just stop passing `webRoutePrefix` as a parameter and we use instead `cpf.RoutePrefix`?
```suggestion
	if cpf.RoutePrefix == "" {
		cpf.RoutePrefix = webRoutePrefix
```

Reply:
I would defer any logic change to a subsequent PR to minimize the risk of regression.